app/src/main/java/com/beemdevelopment/aegis/crypto/otp/YAOTP.java

```diff
@@ -0,0 +1,78 @@
+package com.beemdevelopment.aegis.crypto.otp;
+
+import androidx.annotation.NonNull;
+
+import java.security.InvalidKeyException;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+
+public class YAOTP {
+    private static final int EN_ALPHABET_LENGTH = 26;
+    private final long _code;
+    private final int _digits;
+
+    private YAOTP(long code, int digits) {
+        _code = code;
+        _digits = digits;
+    }
+
+    static private long fromBigEndianArray(int startIndex, byte[] array) {
+        long value = array[startIndex] & 0x7f;
+        for (int i = startIndex + 1; i <= startIndex + 7; i++)
+            value = (value << 8) | array[i] & 0xff;
+        return value;
+    }
+
+    public static YAOTP generateOTP(byte[] secret, byte[] pin, int digits, String otpAlgo, long period)
+            throws NoSuchAlgorithmException, InvalidKeyException {
+        long seconds = System.currentTimeMillis() / 1000;
+        return generateOTP(secret, pin, digits, otpAlgo, seconds, period);
+    }
+
+    public static YAOTP generateOTP(byte[] secret, byte[] pin, int digits, String otpAlgo, long seconds, long period)
+            throws NoSuchAlgorithmException, InvalidKeyException {
+
+        long counter = (long) Math.floor((double) seconds / period);
+
+        // Concating pin and secret together
+        byte[] pinWithHash = new byte[pin.length + secret.length];
+        System.arraycopy(pin, 0, pinWithHash, 0, pin.length);
+        System.arraycopy(secret, 0, pinWithHash, pin.length, secret.length);
```

Comment:
I would prefer using ``ByteArrayOutputStream`` here. It might be a little slower, but it's easier to read.

Reply:
No problem, changed it